aws/terminator/paas.py

```diff
@@ -20,3 +22,121 @@ def ignore(self) -> bool:
+    def age_limit(self):
+        return datetime.timedelta(minutes=20)
+
+    @property
+    def name(self):
+        return self.instance['clusterName']
+
+    @staticmethod
+    def create(credentials):
+        def _paginate_cluster_results(client):
+            names = client.get_paginator('list_clusters').paginate(
+                PaginationConfig={
+                    'PageSize': 100,
+                }
+            ).build_full_result()['clusterArns']
+
+            if not names:
+                return []
+
+            return client.describe_clusters(name=names)['clusters']
+
+        return Terminator._create(credentials, Ecs, 'ecs', _paginate_cluster_results)
+
+    def terminate(self):
+        def _paginate_task_results(container_instance):
+            names = self.client.get_paginator('list_tasks').paginate(
+                cluster=self.name,
+                containerInstance=container_instance,
+                PaginationConfig={
+                    'PageSize': 100,
+                }
+            ).build_full_result()['taskArns']
+
+            if not names:
+                return []
+
+            return self.client.describe_taks(cluster=self.name, tasks=names)['tasks']
+
+        def _paginate_container_instance_results():
+            names = self.client.get_paginator('list_container_instances').paginate(
+                cluster=self.name,
+                PaginationConfig={
+                    'PageSize': 100,
+                }
+            ).build_full_result()['containerInstanceArns']
+
+            if not names:
+                return []
+
+            return self.client.describe_container_instances(cluster=self.name, containerInstances=names)['containerInstances']
+
+        def _paginate_service_results():
+            names = self.client.get_paginator('list_services').paginate(
+                cluster=self.name,
+                PaginationConfig={
+                    'PageSize': 100,
+                }
+            ).build_full_result()['serviceArns']
+
+            if not names:
+                return []
+
+            return self.client.describe_services(cluster=self.name, services=names)['services']
+
+        container_instances = _paginate_container_instance_results()
+        for name in container_instances:
+            # If there are tasks running on a container instance, stop them first
+            tasks = _paginate_task_results(name['containerInstanceArn'])
+            for task in tasks:
+                self.client.stop_task(cluster=self.name, task=task['taskArn'])
+
+        # If there are running services, delete them first
+        services = _paginate_service_results()
+        for name in services:
+            self.client.delete_service(cluster=self.name, service=name['serviceName'], force=True)
+
+        # Deregister container instances
+        for name in container_instances:
+            self.client.deregister_container_instance(containerInstance=name['containerInstanceArn'])
+
```

Comment:
Before deleting the cluster, you will also need to delete all the tasks. In the case of fargate, there are no container instances and you may have tasks running that aren't managed by a service. I'm also not clear on whether using `force=True` when deleting a service also deletes the tasks or just orphans them. Either way, there should be a step that stops all tasks.